<?xml version="1.0" encoding="utf-8"?>
<launch>

  <node pkg="pd3_demo" name="follower_core" type="follower_core.py" output="screen">
    <!-- LiDarと人との目標距離[m] -->
    <param name="target_dist" type="double" value="0.5" />
    <!-- LiDarと人との安全距離[m]. この距離になったら緊急停止する -->
    <param name="safety_dist" type="double" value="0.3" />
    <!-- 人を検出する範囲[px]の設定に使用-->
    <param name="range_xmin"  type="int" value="200" />
    <param name="range_xmax"  type="int" value="300" />
    <param name="range_ymin"  type="int" value="100" />
    <param name="range_ymax"  type="int" value="250" />
    <!-- PublishするTwist型のトピック名 -->
    <!-- <param name="pub_twist_name"  type="str" value="/cmd_vel" /> -->
    <param name="pub_twist_name"  type="str" value="/mobile_base/commands/velocity" />
    <!-- 並進PID制御の各ゲイン -->
    <param name="lKp"         type="double" value="0.056" />
    <param name="lKi"         type="double" value="0.00" />
    <param name="lKd"         type="double" value="0.00" />
    <!-- 旋回PID制御の各ゲイン -->
    <param name="aKp"         type="double" value="0.017" />
    <param name="aKi"         type="double" value="0.0" />
    <param name="aKd"         type="double" value="0.0" />
    <!-- Twistの最大,最小速度 -->
    <param name="max_linear"  type="double" value="0.2" />
    <param name="min_linear"  type="double" value="0.0" />
    <param name="max_angular" type="double" value="1.0" />
    <param name="min_angular" type="double" value="-1.0" />
  </node>

  <node pkg="pd3_demo" type="follower_imgshow.py" name="follower_imgshow" output="screen"/>

</launch>
